benchmarks/benchmarks.py

```diff
@@ -0,0 +1,151 @@
+# -*- coding: utf-8 -*-
+"""
+Fonctions testant le temps d'execution de chacune des taches du core
+"""
+
+import timeit
+from datetime import datetime, timedelta
+import random
+from cide.app.python.core import Core
+import logging
+import gc
+from Queue import Queue
+from pdb import set_trace as debug
+from libZoneTransit import Addition
+
+class Benchmarks(object):
+  def __init__(self):
+      
+    self.project_conf = dict(name = 'Benchmarks',
+                base_dir = './temp',
+                code_dir = './temp',
+                exec_dir = './temp',
+                backup_dir = './temp',
+                tmp_dir = './temp')
+    
+    #Not used since tasks are called directly              
+    self.core_conf = dict(cycle_time = 0.0, 
+                 buffer_critical = 0.0,
+                 buffer_secondary = 0.0,
+                 buffer_auxiliary = 0.0)
+                 
+    # Setup Log
+    formatter = logging.Formatter('%(asctime)s - %(levelname)s - %(module)s - %(message)s')
+    handler = logging.FileHandler("log_benchmarks")
+    handler.setFormatter(formatter)
+    self.logger = logging.getLogger('cide.py')
+    self.logger.setLevel(logging.DEBUG)
+    self.logger.addHandler(handler)
+    
+    self.core = None
+    self.callers = []
+    self.files = []
+    self.changes = []
+    
+    for i in range(50):
+        self.callers.append(chr(ord('a')+i / 26) + chr(ord('a')+i % 26))
+        filename = ""
+        for j in range(random.randint(1,100)):
+            filename += '/' + chr(ord('a')+i / 26) + chr(ord('a')+i % 26)
+        self.files.append(filename)     
```

Comment:
Should include constants from core.py (or a config file) to control the number of users and files